src/utils/hooks/useSearchProducts.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { setPage, setTotalPages } from '../../actions/page';
+import { setFeaturedProducts } from '../../actions/products';
+import { API_BASE_URL } from '../constants';
+import { filterProducts } from '../selectors/filterProducts';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useSearchProducts(searchTerm, page) {
```

Comment:
You need to think of a way to reuse the hook to fetch the data from the API, right now, there is a hook for each call, thinking in reusability and abstraction.